code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -58,6 +59,11 @@
 	storages.Cut()
 	return ..()
 
+///Passes the safe decon signal to individual tools, in case any have special cases for such
+/obj/item/robot_module/proc/safedecon()
+	for(var/obj/i in basic_modules+emag_modules+ratvar_modules)
+		SEND_SIGNAL(i, COMSIG_BORG_SAFE_DECONSTRUCT, src)

```

Comment:
Don't do this. Either register the signal from the items directly or create a new proc that' called on_safe_decon or something. First option would be prefered.

Reply:
Signals have been improved.